cmd/deck/pluginhelp.go

```diff
@@ -52,6 +59,27 @@ func (ha *helpAgent) getHelp() (*pluginhelp.Help, error) {
 		return ha.help, nil
 	}
 
+	//Setting Root CAs if SSL is enabled
+	hookPath, err := url.Parse(ha.path)
+	if err != nil {
+		return nil, fmt.Errorf("error parsing hook path: %w", err)
+	}
+	if hookPath.Scheme == tlsEnabledScehma {
+		caCert, err := os.ReadFile(ha.cert)
+		if err != nil {
+			return nil, fmt.Errorf("error decoding cert file: %w", err)
+		}
+		caCertPool := x509.NewCertPool()
+		caCertPool.AppendCertsFromPEM(caCert)
```

Comment:
I am not entirely sure I like this coupling. This essentially reuses Deck's server cert as a CA cert to establish trust with Hook's server cert for this behavior where Deck is a client to hook's pluginhelp endpoint? That's a bit hacky and suprising.